app/queries/find_exact_query.rb

```diff
@@ -0,0 +1,32 @@
+class FindExactQuery
+  class InvalidFindCriteria < StandardError; end
+
+  def initialize(tags:, links:, document_type:, email_document_supertype:, government_document_supertype:)
+    @tags = tags.symbolize_keys
+    @links = links.symbolize_keys
+    @document_type = document_type
+    @email_document_supertype = email_document_supertype
+    @government_document_supertype = government_document_supertype
+  end
+
+  def exact_match
+    if @links.any? || @tags.any?
+      find_exact(:links, @links) || find_exact(:tags, @tags)
```

Comment:
I think this would be a bit clearer if we split `@tags` into an `elsif` - as it is, if there are tags but no links we'll call `find_exact` for links unnecessarily.